sql/src/main/java/io/crate/protocols/postgres/PostgresWireProtocol.java

```diff
@@ -364,9 +364,13 @@ private void handleStartupBody(ByteBuf buffer, Channel channel) {
     private void authenticate(Channel channel, Properties properties) {
         String userName = properties.getProperty("user");
         InetAddress address = CrateNettyHttpServerTransport.getRemoteAddress(channel);
-        boolean usesSSL = channel.pipeline().first() instanceof SslHandler;
-        HbaProtocol hbaProtocol = usesSSL ? HbaProtocol.POSTGRES_SSL : HbaProtocol.POSTGRES;
-        AuthenticationMethod authMethod = authService.resolveAuthenticationType(userName, address, hbaProtocol);
+
+        SslHandler sslHandler = channel.pipeline().get(SslHandler.class);
+        ConnectionProperties connProperties = new ConnectionProperties(sslHandler);
```

Comment:
Maybe the address and protocol should also become part of the connectionProperties?